crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -524,6 +524,260 @@ async fn c8y_mapper_child_alarm_with_custom_fragment_mapping_to_c8y_json() {
+    mqtt_tests::assert_received_all_expected(
+        &mut messages,
+        TEST_TIMEOUT_MS,
+        &["{\"severity\":\"MAJOR\",\"type\":\"msg_to_text_pressure_alarm\",\"time\":\"2023-01-25T18:41:14.776170774Z\",\"text\":\"msg_to_text_pressure_alarm\",\"message\":\"Pressure high\"}"],
+    )
+    .await;
+
+    //Clear the previously published alarm
+    broker
+        .publish_with_opts(
+            "tedge/alarms/major/msg_to_text_pressure_alarm",
+            "",
+            mqtt_channel::QoS::AtLeastOnce,
+            true,
+        )
+        .await
+        .unwrap();
+
+    sm_mapper.abort();
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[serial]
+async fn c8y_mapper_child_alarm_empty_payload() {
+    let broker = mqtt_tests::test_mqtt_broker();
+
+    let mut messages = broker
+        .messages_published_on("c8y/s/us/external_sensor")
+        .await;
+    let cfg_dir = TempTedgeDir::new();
+    // Start the C8Y Mapper
+    let (_tmp_dir, sm_mapper) = start_c8y_mapper(broker.port, &cfg_dir).await.unwrap();
+
+    broker
+        .publish_with_opts(
+            "tedge/alarms/major/empty_temperature_alarm/external_sensor",
+            r#"{}"#,
+            mqtt_channel::QoS::AtLeastOnce,
+            true,
+        )
+        .await
+        .unwrap();
+
+    let expected_msg = r#"302,empty_temperature_alarm"#;
+
+    while let Ok(Some(msg)) = messages.next().with_timeout(TEST_TIMEOUT_MS).await {
+        assert!(msg.contains(expected_msg));
+    }
+
+    //Clear the previously published alarm
+    broker
+        .publish_with_opts(
+            "tedge/alarms/major/empty_temperature_alarm/external_sensor",
+            "",
+            mqtt_channel::QoS::AtLeastOnce,
+            true,
+        )
+        .await
+        .unwrap();
+
+    sm_mapper.abort();
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[serial]
+async fn c8y_mapper_alarm_empty_payload() {
+    let broker = mqtt_tests::test_mqtt_broker();
+
+    let mut messages = broker.messages_published_on("c8y/s/us").await;
+    let cfg_dir = TempTedgeDir::new();
+    // Start the C8Y Mapper
+    let (_tmp_dir, sm_mapper) = start_c8y_mapper(broker.port, &cfg_dir).await.unwrap();
+
+    broker
+        .publish_with_opts(
+            "tedge/alarms/major/empty_pres_alarm",
+            r#"{}"#,
+            mqtt_channel::QoS::AtLeastOnce,
+            true,
+        )
+        .await
+        .unwrap();
+
+    // Expect converted temperature alarm message
+    mqtt_tests::assert_received_all_expected(
+        &mut messages,
+        TEST_TIMEOUT_MS,
+        &["302,empty_pres_alarm"],
+    )
+    .await;
+
+    //Clear the previously published alarm
+    broker
+        .publish_with_opts(
+            "tedge/alarms/major/empty_pres_alarm",
+            "",
+            mqtt_channel::QoS::AtLeastOnce,
+            true,
+        )
+        .await
+        .unwrap();
+
+    sm_mapper.abort();
+}
```

Comment:
Another test case worth adding: Mapping of a tedge alarm with a non-string type or complex JSON fragment in the `text` field. If C8y also supports complex JSON values in the `text` field, then it should be okay. Else, the mapping should fail with that message.

Reply:
Added a test with complex JSON object in the text field.